src/test/java/org/mule/test/oauth/AbstractOAuthTestCase.java

```diff
@@ -73,13 +82,33 @@ public void setupServices() throws Exception {
     when(httpServerFactory.create(any())).thenReturn(httpServer);
     when(httpService.getServerFactory()).thenReturn(httpServerFactory);
 
-    service = new DefaultOAuthService(httpService, mock(SchedulerService.class));
+    service = new DefaultOAuthService(httpService, new SimpleUnitTestSupportSchedulerService());
 
     final HttpResponse httpResponse = mock(HttpResponse.class);
     final InputStreamHttpEntity httpEntity = mock(InputStreamHttpEntity.class);
     when(httpEntity.getContent()).thenReturn(new ReaderInputStream(new StringReader("")));
     when(httpResponse.getEntity()).thenReturn(httpEntity);
-    when(httpClient.sendAsync(any(), any())).thenReturn(supplyAsync(() -> httpResponse));
+    when(httpClient.sendAsync(any(), any())).thenAnswer(invocation -> {
+
+      final CompletableFuture<HttpResponse> httpResponseFuture = new CompletableFuture<>();
+
+      httpClientCallbackExecutor.execute(() -> {
+        try {
+          sleep(10);
```

Comment:
didn't we said in the past review that 10 was too short and should use 100 instead?

Reply:
yep, in another piece of code. This is the test case